Dockerfile

```diff
@@ -1,52 +1,38 @@
-FROM rust:1.67 as build-env
+FROM debian:11 as build-env
+
 WORKDIR /src
 
-RUN apt-get update &&\
-    apt-get install -y libssl-dev texinfo libcap2-bin &&\
-    apt-get clean && rm -rf /var/lib/apt/lists/*
+# Copy all the source files
+# .dockerignore ignores the target dir
+# This includes the rust-toolchain.toml
+COPY . .
+
+# Install dependencies
+RUN apt-get update && \
+    apt-get install -y curl build-essential libssl-dev texinfo libcap2-bin pkg-config
 
-ARG BIN=rust-app
+# TODO: Use a specific version of rustup
+# Install rustup
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
```

Comment:
I'm installing rustup here instead of using a rust base container, because I want to be able to only set the versino in `rust-toolchain.toml`